scene/main/viewport.cpp

```diff
@@ -1948,6 +1957,33 @@ void Viewport::_gui_input_event(Ref<InputEvent> p_event) {
 				gui.drag_data=Variant(); //always clear
 			}*/
 
+			// in case the mouse was released after for example dragging a scrollbar,
+			// check whether the current control is different from the stored one. If
+			// it is different, rather than wait for it to be updated the next time the
+			// mouse is moved, notify the control so that it can e.g. drop the highlight.
+			// This code is duplicated from the mm.is_valid()-case further below.
+			if (gui.mouse_focus) {
+				over = gui.mouse_focus;
+				//recompute focus_inv_xform again here
```

Comment:
Is this a TODO?

Reply:
I am not sure. It is exactly the code that is used to update the current hover on mousemove-events further down. 

https://github.com/godotengine/godot/blob/430460b94b76524cd9a563398320e29135dc3acd/scene/main/viewport.cpp#L2040-L2069

Maybe that comment should be removed for the duplicated version.

Reply:
That comment in the original section that I copied has been around since 0b806ee0fc9097fa7bda7ac0109191c9c5e0a1ac so who knows what the intention behind the comment was.

Reply:
Maybe is it time to remove it? An additional benefit would be that the diff will show a change in the original instance of the code, and that would help with traceability.